evalml/automl/automl_search.py

```diff
@@ -358,7 +358,12 @@ def __init__(self,
             self._sampler_name = self.sampler_method
             if self.sampler_method in ['auto', 'Oversampler']:
                 self._sampler_name = get_best_sampler_for_data(self.X_train, self.y_train, self.sampler_method, self.sampler_balanced_ratio)
-            self._frozen_pipeline_parameters[self._sampler_name] = {"sampling_ratio": self.sampler_balanced_ratio}
+            if self._sampler_name not in parameters:
+                parameters[self._sampler_name] = {"sampling_ratio": self.sampler_balanced_ratio}
+                self._frozen_pipeline_parameters[self._sampler_name] = {"sampling_ratio": self.sampler_balanced_ratio}
+            elif self._sampler_name in parameters:
+                parameters[self._sampler_name].update({"sampling_ratio": self.sampler_balanced_ratio})
```

Comment:
Decided to default to the `sampler_balanced_ratio` value that was provided for AutoMLSearch to set the `sampling_ratio`, which means it overrides the `sampling_ratio` that a user might input to the pipeline_params via
```python
pipeline_parameters = {sampler : {"sampling_ratio": 0.5}}
```

Updated the docstring above to mention this

Reply:
Very cool! I don't think you need an `elif` here, it can just be an `else` since I don't think we'd have another logic path but total nitpick.

Also I like the way you set the logic for the `elif` block. You could do the same to the `if` block and move setting the `self._frozen_pipeline_parameters` after:
```
if self._sampler_name not in parameters:
     parameters[self._sampler_name] = {"sampling_ratio": self.sampler_balanced_ratio}
else:
     parameters[self._sampler_name].update({"sampling_ratio": self.sampler_balanced_ratio})
self._frozen_pipeline_parameters[self._sampler_name] = parameters[self._sampler_name]
```